pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -102,6 +108,28 @@ func (ts *TaskSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateUsageOfDeclaredParameters validates that all parameters referenced in the Task are declared by the Task.
+func validateUsageOfDeclaredParameters(ctx context.Context, steps []Step, params ParamSpecs) *apis.FieldError {
```

Comment:
Even though this function is not exported, it is called in pipeline and task validation. Do you think it should have its own unit test?

Reply:
That's a good question-- it would probably be nice to have unit tests for this function, and we definitely aim to test each function individually. However this function already has test coverage from existing tests for taskSpec.Validate, and task_validation_test.go has package v1beta1_test. I'm going to export this in a subsequent PR, maybe it would make sense to add tests for this function specifically then?

Reply:
Yes, that's fine!